### ISSUE TITLE
Exception in processing messages

### ISSUE THREAD

**Issue by lpaatero** [moquette-io/moquette#441]
### Expected behavior

No exceptions when processing normal messages.

### Actual behavior
```
Exception is thrown:
java.util.NoSuchElementException
        at java.util.AbstractQueue.remove(AbstractQueue.java:117)
        at io.moquette.broker.Session.drainQueueToConnection(Session.java:319)
        at io.moquette.broker.Session.pubAckReceived(Session.java:274)
        at io.moquette.broker.MQTTConnection.processPubAck(MQTTConnection.java:124)
        at io.moquette.broker.MQTTConnection.handleMessage(MQTTConnection.java:90)
        at io.moquette.broker.NewNettyMQTTHandler.channelRead(NewNettyMQTTHandler.java:61)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.moquette.broker.metrics.MQTTMessageLogger.channelRead(MQTTMessageLogger.java:46)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.moquette.broker.metrics.MessageMetricsHandler.channelRead(MessageMetricsHandler.java:50)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:310)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:284)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
```
### Steps to reproduce

It seems that !canSkipQueue is not sufficient condition to ensure that sessionQueue contains elements in while loop in drainQueueToConnection():
```java
  while (!canSkipQueue()) {
    final SessionRegistry.EnqueuedMessage msg = sessionQueue.remove();
```
Looking at canSkipQueue this seems to be the case: 
```java
  return sessionQueue.isEmpty() && inflightSlots.get() > 0 && connected() && mqttConnection.channel.isWritable();
```

### Moquette MQTT version

0.12

**Comment by rpahli** [comment]
The Problem is the inflight check. I had the same problem I increased  the number (default  in 10). It would be good to have a config option for it.

**Comment by andsel** [comment]
I think the problem is that when a PUBACK (for QoS1 publish) comes from the subscriber to the broker and the queue is empty, we got the error. I'm fixing it

**Comment by andsel** [comment]
Fixed by 3f7c8fe3ea6761d2617318a7d910d99a60362e23
